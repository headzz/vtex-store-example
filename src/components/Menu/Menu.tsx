import { useState } from 'react'
import Burguer from './components/Burguer/Burguer'
import MenuItems from './components/MenuItems/MenuItems'

/**
 * @component
 * @name Menu
 * @description Defines Menu component for mobile.
 * @namespace components components/Menu
 *
 */
const Menu = () => {
  const [open, setOpen] = useState<boolean>(false)
  return (
    <>
      <Burguer open={open} setOpen={setOpen} />
      <MenuItems open={open} />
    </>
  )
}

export default Menu
