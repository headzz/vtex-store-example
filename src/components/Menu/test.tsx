import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import Menu from './Menu'

describe('Menu', () => {
  it('should render Menu', () => {
    const { container } = render(<Menu />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should show option', () => {
    render(<Menu />)
    userEvent.click(screen.getByRole('button'))
    expect(screen.getByText('Login')).toBeInTheDocument()
  })
})
