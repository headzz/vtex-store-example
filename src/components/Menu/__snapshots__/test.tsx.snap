// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should render Menu 1`] = `
.c0 {
  position: fixed;
  top: 2.4rem;
  right: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  width: 3rem;
  height: 3rem;
  background: transparent;
  border: none;
  cursor: pointer;
  padding: 0;
  z-index: 2;
}

.c0:focus {
  outline: none;
}

.c0 div {
  width: 3rem;
  height: 0.2rem;
  background: #0d0c1d;
  border-radius: 10px;
  -webkit-transition: all 0.3s linear;
  transition: all 0.3s linear;
  position: relative;
  -webkit-transform-origin: 1px;
  -ms-transform-origin: 1px;
  transform-origin: 1px;
}

.c0 div:first-child {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

.c0 div:nth-child(2) {
  opacity: 1;
}

.c0 div:nth-child(3) {
  -webkit-transform: rotate(0);
  -ms-transform: rotate(0);
  transform: rotate(0);
}

@media (min-width:768px) {
  .c0 {
    display: none;
  }
}

@media (min-width:768px) {

}

<button
  class="c0"
>
  <div />
  <div />
  <div />
</button>
`
